src/block/block_map.c

```diff
@@ -24,7 +24,7 @@ __wt_block_map(WT_SESSION_IMPL *session, WT_BLOCK *block, void *mapped_regionp,
     *(void **)mapped_cookiep = NULL;
 
     /* Map support is configurable. */
-    if (!S2C(session)->mmap)
+    if (!S2C(session)->mmap_ckpt)
```

Comment:
Agree. That would entail getting rid of synchronization for the read-only case. The synchronization that's in the branch is only there, because one session can change the file size while the other is accessing the same file via mmap.  In read-only scenarios this cannot happen, so the synchronization is not needed. We can add a few if-statements to elide synchronization is the file is mapped readonly.

Reply:
I won't make changes to merge these two cases now, as this seems to be a rather major change. Perhaps later, once you've made your pass.

Reply:
@keithbostic, I will mark this as resolved, and let's open a separate ticket to address this once this branch is merged. Please reopen if you disagree.